action.yml

```diff
@@ -20,6 +20,18 @@ inputs:
     required: false
     default: 'all'
     description: 'The type of tests to be run by the test runner.'
+  enableCodeCoverage:
+    required: false
+    default: 'false'
+    description: 'Should code coverage results be collected for the test run.'
+  coverageAssemblyFilters:
+    required: false
+    default: ''
+    description: 'Optional coverage assembly filters when collecting code coverage.'
+  coverageResultsPath:
+    required: false
+    default: 'CodeCoverage'
+    description: 'Path to write coverage results to when collecting code coverage.'
```

Comment:
Would it be possible to follow convention (folder is usually called `[Cc]overage` from projects across a few languages I've seen).

Perhaps it could be stored inside the `artifacts` folder that we already create containing the test results (many people have already ignored this in their `.gitignore`)?

Reply:
Sure, adding it to the `[Cc]overage` folder (Unity's project's default folder is called `CodeCoverage`) as a default arugmnet is a great idea. The one reason I would try to avoid putting it in the `artifacts` is that it can be a fairly large build. We can make the default `coverage` but let the user configure them as different path (even `artifacts` if they want) to allow them to control each export individually.

Reply:
Ah if Unity has `CodeCoverage` as a default, then we should probably follow their convention.

> The one reason I would try to avoid putting it in the `artifacts` is that it can be a fairly large build.

`unity-builder` doesn't output to `artifacts` folder though. `unity-test-runner` currently puts all it's outputs in `artifacts` folder. It's not more than a few megs large.

If with that information you still think it's a bad idea to expose it as part of the existing output folder, then I suggest we hardcode `CodeCoverage` for the folder and `output.coverage` for the output parameter in favour of "great defaults over many configuration options", at least for now.

Let me know what you think.

Reply:
I agree with this, let's keep the default the same as unity's for now and not make it configurable (at least for this version). I'll remove the input parameter but keep the output parameter. Unity has UI for configuring the code coverage so let's use that as a guide for what we should make as all the configurable parameters, although we can leave that to a future PR.

![image](https://user-images.githubusercontent.com/3240136/164141052-800c7db9-e12c-4c3e-8d3a-5ff43d9478f7.png)

They have the configurable parameters:
* `Results Location`: output location which we can make configurable later if necessary... It defaults to `CodeCoverage` so we can leave that as the output path for now and add it as an input parameter if we need. 
* `History Location`: which is where the historical results of tests are saved, not as useful for reporting so we can probably leave it out for now.
* `Enable Code Coverage`: which we can probably always leave enabled as it doesn't hurt anything in the test build. If an end user wants to run a perf test, they should probably make a compiled build first anyway so not harm in including the debug flag. 
* `Included Assemblies`: This is the assembly filter argument; this could probably be made this something a user could edit by the existing `coverageParameters` argument.
* `Included Paths`/`Excluded Paths`: can also be configured from the `coverageParameters` argument. Slightly complex for an end user but it will default to all the code included in the `Assets` folder so unless they included custom libraries, they should be fine.
* `Generate HTML Report`: This is very useful for visually seeing the coverage but is quite a large file set if the user has a large project or lots of imported libraries. So, this might be a flag we want to add in the future. Although this can also be configured from the from the `coverageParameters` argument as well so maybe just leave it for more advanced users if they want. 
* `Generate Summary Badges`: This generates the badges for showing whether it passed or not as an icon, fairly lightweight so we can leave it always enabled.
* `Generate History`: Again, history is useful for debugging but not as useful in reporting on a single build so we can leave it disabled. 
* `Generate Additional Metrics`: I think this is something most users wouldn't need but could be added with the `coverageParameters` argument if the user wants to.


Sorry for the long explanation, but basically I suggest we add these parameters now:
* `coverageParametrers` for the custom coverage parameters via the `-coverageOptions` that allows us to configure most all of the code coverage settings. 
* Remove the `coverageResultsPath` as a default of `[Project]/CodeCoverage` should be sufficient for most users and aligns with the unity default. 
* Keep the output path of results as you suggested in case we add configuration in the future!

With just adding one parameter for `coverageParametrers` (which I will rename to `coverageOptions` to align with the flag name), the user can configure almost everything that they would want to! The parameter would be a bit difficult to configure sometimes but most users probably want the HTML report or could ignore it in their file upload if they don't want to save it.